### ISSUE TITLE
(Linux-only) Unexpected results when using UC Mode while changing "user-data-dir"

### ISSUE THREAD

**Issue by M-Zubair10** [seleniumbase/SeleniumBase#1758]
When I open seleniumbase using Driver class with uc=True, page_load_strategy='none', headless2=True, and user_data_dir=data_directory, it opens chrome even headless2=True, also there is flag active named controlled by automated driver
and this happens only when I use data_directory, unfortunately I need to use data_dir

Moreover, undetected_chromedriver and selenium vanilla works fine

I am now in Linux, works perfect in windows

**Comment by mdmintz** [comment]
Given the number of open issues involving changing the ``user-data-dir`` while in UC Mode (https://github.com/ultrafunkamsterdam/undetected-chromedriver/search?q=user-data-dir&type=issues), there isn't much I can do unless there's an upstream change to fix those in ``undetected-chromedriver`` directly. You're best option is to make your tests work without changing the ``user-data-dir``.

**Comment by M-Zubair10** [comment]
But i am not changing data dir, it always the same i.e abspath(./chrome-1)

**Comment by mdmintz** [comment]
You're trying to set a ``user-data-directory``. Use the default one that chromedriver uses for tests.

**Comment by M-Zubair10** [comment]
Default one does not load tampermonkey scripts, and also default dir does
not save all cookies that I need
By the way it works on windows, can you check please

**Comment by mdmintz** [comment]
You can try with ``pytest --xvfb`` on Linux.
You can also try upgrading SeleniumBase to ``4.13.0``.

**Comment by M-Zubair10** [comment]
Seleniumbase is already on latest version, moreover why to run pytest, just
fix the vanilla

**Comment by mdmintz** [comment]
There's no way you could have run a test that fast. I just released a new version to PyPI.

<img width="1285" alt="Screenshot 2023-02-18 at 2 45 11 AM" src="https://user-images.githubusercontent.com/6788579/219848504-d3e63458-b79c-40ce-bb30-e97722281a89.png">

And ``pytest`` includes options that you can't get with the raw driver, such as ``--xvfb``, which you may need on Linux.

**Comment by M-Zubair10** [comment]
You are right, actually i updated it when I make this issue

**Comment by M-Zubair10** [comment]
Tried new version but still same issue

**Comment by mdmintz** [comment]
I can confirm that things work as expected when setting a ``user-data-dir`` and running SeleniumBase scripts in UC Mode when using ``pytest`` in order to get the additional options, such as automatic Xvfb initialization when running on Linux systems:

https://github.com/mdmintz/undetected-testing/actions/runs/4214029106

Here's the GitHub Actions yml file:
https://github.com/mdmintz/undetected-testing/blob/master/.github/workflows/python-package-2.yml

**Comment by mdmintz** [comment]
Looks like the issue was intermittent in CI, and I was able to see it this time:

https://github.com/mdmintz/undetected-testing/actions/runs/4223965774/jobs/7334317012

(When trying to use an existing User Data Dir.)

```bash
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/seleniumbase/core/browser_launcher.py:2873: in get_local_driver
    driver = undetected.Chrome(
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/seleniumbase/undetected/__init__.py:243: in __init__
    options.handle_prefs(user_data_dir)
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/site-packages/seleniumbase/undetected/options.py:54: in handle_prefs
    undot_prefs = self._merge_nested(json.load(f), undot_prefs)
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/json/__init__.py:293: in load
    return loads(fp.read(),
/opt/hostedtoolcache/Python/3.10.10/x64/lib/python3.10/json/__init__.py:346: in loads
    return _default_decoder.decode(s)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x7fd909d96bc0>
s = '{"NewTabPage": {"PrevNavigationTime": "13321[37](https://github.com/mdmintz/undetected-testing/actions/runs/4223965774/jobs/7334317012#step:11:38)4088316310"}, "account_tracker_service_last_update": "13321374088514[38](https://github.com/mdmintz/undetected-testing/actions/runs/4223965774/jobs/7334317012#step:11:39)6...ERY\\",\\"QUERY\\",\\"ENTITY\\",\\"QUERY\\",\\"QUERY\\",\\"QUERY\\"],\\"google:verbatimrelevance\\":851}]"}}":851}]"}}'
_w = <built-in method match of re.Pattern object at 0x7fd909e920c0>

    def decode(self, s, _w=WHITESPACE.match):
        """Return the Python representation of ``s`` (a ``str`` instance
        containing a JSON document).
    
        """
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
        end = _w(s, end).end()
        if end != len(s):
>           raise JSONDecodeError("Extra data", s, end)
E           json.decoder.JSONDecodeError: Extra data: line 1 column 38701 (char 38700)
```

I think I might be able to fix it this time. It'll be part of the next release.

**Comment by mdmintz** [comment]
It was caused by https://stackoverflow.com/questions/21058935/python-json-loads-shows-valueerror-extra-data
(A very popular issue with over 500k views)

**Comment by M-Zubair10** [comment]
Haha
I actually switched my web server from linux to windows
So I no longer need this
On windows, it works as it should

**Comment by mdmintz** [comment]
This was fixed in ``4.13.1`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.13.1

**Comment by mdmintz** [comment]
There were a few more changes for this in ``4.13.2`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.13.2
